### ISSUE TITLE
Unable to create Static Web App for Hugo

### ISSUE THREAD

**Issue by jezkerwin** [Azure/static-web-apps#274]
I'm trying to create a static app via the portal and using the Hugo build template.
Following the portal setup, I sign into my github account and successfully authorise the app access to my repos which I'm trying to use a private repo.
I'm also using a Pay as I go Subscription

The build is failing with the following error.

{
    "status": "Failed",
    "error": {
        "code": "InternalServerError",
        "message": "There was an unexpected InternalServerError.  Please try again later.  x-ms-correlation-request-id: d9d64a4c-cb19-4f82-b950-2ad8b5dc8f1e"
    }
}

I've seen some posts talk about a personal access token in github but it doesn't appear to be creating when I 'sign into github' using the portal page.

**Comment by miwebst** [comment]
Hi @jezkerwin do you mind providing the region and time frame when you saw the above creation failure? I can help identify the problem from there!

**Comment by jezkerwin** [comment]
thanks @miwebst I just did another task and it failed again.
Region was US West 2 Time I did the task was 1/26/2021 21:30UTC

Operations ID: /subscriptions/263bfd25-9f10-4b6f-8814-ff7917e47d58/resourceGroups/astronerd/providers/Microsoft.Resources/deployments/Microsoft.Web-StaticApp-Portal-93c30b38-8792/operations/283EAF5FEF1915CA
Tracking ID : c4419578-292d-4fc6-bdf7-e0b84e7f3b83
Service Request ID: c4419578-292d-4fc6-bdf7-e0b84e7f3b83

**Comment by miwebst** [comment]
So we are failing to add the workflow file to your GitHub repository. Are there are policies on your repo that restrict commits? similar to this: https://github.com/Azure/static-web-apps/issues/241

**Comment by jezkerwin** [comment]
Not that I'm aware of. I've created just a standard private repo, not made any changes to it and pushed my code to it.
I haven't adjusted any policies

**Comment by jezkerwin** [comment]
I've made some progress. I delete my github repo and created a blank one with just a readme file.
Deployed the Azure static web app and the deployment succeeded.

After I perform a git push from my workstation in VS Code it seems that it locks the repo out to be able to create any files others than the git push.

Even in github.com if I try and click "Create a File" I get an error saying 'File could not be created". I have no idea what my git client is doing.

**Comment by jezkerwin** [comment]
I'm closing out this problem because I figured out the problem. 
Because I'm building a Hugo site I followed their instructions to added a theme as a git submodule.
Turns out the .gitmodules file was causing the issues. I remove the file and everything appears to be working ok, azure static web app deploys successfully.
I'll just have to figure out why .gitmodules is preventing creating files in the repo.
